### ISSUE TITLE
Add-on review submission is blocked with a "...can't leave more than one review for the same add-on..." message

### ISSUE THREAD

**Issue by ValentinaPC** [mozilla/addons-frontend#2019]
Steps to reproduce:
1. Load AMO-dev details page for an add-on that wasn't yet reviewed by you (on your device)
2. Tap some review stars but change you mind by clicking outside the review submission popup
3. Refresh the page
4. Try to add a review for the same add-on

Expected results:
Review submission popup is displayed.

Actual results:
A message saying that you cannot add 2 reviews for the same add-on is displayed.

Notes/Issues:
- after step 2 the review counter is incremented (even if no review was submitted)
- at refresh rating star, selected at step 2, disappears
- second review for the same add-on should load edit review popup 

Verified on AMO-dev FF51(Android 6.0.1). Issue is reproducing on -dev and -stage.
Video for this issue:
![videotogif_2017 03 08_18 07 04](https://cloud.githubusercontent.com/assets/15685960/23712452/a3d663a0-042b-11e7-8b28-822e481f7a23.gif)

**Comment by kumar303** [comment]
I can't reproduce this. When I refresh the page, the saved star rating is there. Can you reproduce it every time? Could you attach a console log when it happens?

**Comment by ValentinaPC** [comment]
I can reproduce this every time on device or "mobile view" for desktop.
Please see the screenshot for the logs:
![2017-03-09_1022](https://cloud.githubusercontent.com/assets/15685960/23741690/71329752-04b2-11e7-8e99-af1d775ae44b.png)

**Comment by kumar303** [comment]
Thanks for the console log, that's helpful. It looks like some kind of API call is failing but I can't find the error in our server logs. 

Does this happen for every add-on? Can you provide an example of an add-on it happens for on dev?

Also, your STR doesn't mention how you first logged in. Did you do it with the 'log in to rate this add-on' or before?

**Comment by ValentinaPC** [comment]
Your welcome!

> Also, your STR doesn't mention how you first logged in. Did you do it with the 'log in to rate this add-on' or before?

I was already logged. Login was made normally, from "Log in /Sign up" button.

> Does this happen for every add-on? Can you provide an example of an add-on it happens for on dev?

You must use and add-on you did not reviewed before. It must be your first time. 
And it is mandatory to refresh the page after dismissing first attempt to review. (as in step 3)
Happens for me for any add-on.

Examples of add-ons:
- https://addons-dev.allizom.org/en-US/android/addon/youtube-10423/
- https://addons-dev.allizom.org/en-US/android/addon/3-maps/
- https://addons-dev.allizom.org/en-US/android/addon/pdfjs/
- etc.

You can try on desktop too, while being in "View Mobile Site" mode.

**Comment by kumar303** [comment]
Thanks! I can reproduce it now.

**Comment by kumar303** [comment]
A note for myself: it makes no difference if you enter review text or not before hitting refresh. The stars are not saved either way.

**Comment by kumar303** [comment]
Root cause: https://github.com/mozilla/addons-server/issues/5006

**Comment by diox** [comment]
@kumar303 I merged my workaround for https://github.com/mozilla/addons-server/issues/5072, adding `version` filtering. So, when you're trying to find out if the user has already posted a review on the current addon (and only that - don't change anything to the call fetching the list of reviews to the add-on), you should filter on `user`, `addon` and `version` (using the id of the current version of the addon) at the same time.

Note that this will effectively allow the user to post a new review if the addon current version changed since they posted a review in the past. This is fine and expected to be possible.

**Comment by ValentinaPC** [comment]
Verified this on FF53(Android 6.0.1) for add-ons and themes.
- the message saying that you cannot leave 2 reviews for an add-on is no longer displayed.
- rating stars initially selected are listed
- _number of reviews is incremented - expected?_
- no review in "All written reviews" page

**Comment by kumar303** [comment]
> number of reviews is incremented - expected?

Yes because when you select a star it technically records a review, just one without any text

> no review in "All written reviews" page

I'm not sure what you mean. Could you clarify the exact steps and what the expectation was? If you don't submit any text then it shouldn't show up in All written reviews. If you think it's a bug, could you file a new one?

**Comment by ValentinaPC** [comment]
> I'm not sure what you mean. Could you clarify the exact steps and what the expectation was? If you don't submit any text then it shouldn't show up in All written reviews. If you think it's a bug, could you file a new one?

I'm not sure it is, but, I think, same behavior should apply in both mobile and desktop site.

While, not entering a review on device, but the rating stars, two things are happening:
- **on device** the number of reviews, under the ratings stars, is incremented, even if the review process was dismissed - this affects the calculation of average raring stars, and I don't know if is ok...
![device](https://cloud.githubusercontent.com/assets/15685960/25371371/31dddbb4-2999-11e7-96d8-e01fa75ad7d0.png)

- **on desktop**, the review is recorded, containing only the rating stars. (this is an example  https://addons-dev.allizom.org/en-US/android/addon/lock/reviews/ )
![desktop](https://cloud.githubusercontent.com/assets/15685960/25371377/380148b4-2999-11e7-949d-ee4ab9f2ed86.png)

**Comment by ValentinaPC** [comment]
> While, not entering a review on device, but the rating stars, two things are happening:
> 
>     on device the number of reviews, under the ratings stars, is incremented, even if the review process was dismissed - this affects the calculation of average raring stars, and I don't know if is ok...

@kumar303 : is this intended?

Because while dismissing a review (not submitting an empty one), it is registered in desktop add-on details page in the header section under a link, but not in the Reviews section. 
Please see screenshot:
![rev](https://cloud.githubusercontent.com/assets/15685960/25949503/61fc465e-3660-11e7-9c6f-3c9d78f9276d.png)

Please let me know if this require a separate issue or we can use this one. Thanks!

**Comment by kumar303** [comment]
> on device the number of reviews, under the ratings stars, is incremented, even if the review process was dismissed

Ah, good catch. This is a known bug and will be fixed in https://github.com/mozilla/addons-frontend/issues/2328

> on desktop, the review is recorded, containing only the rating stars. 

This should have been fixed in https://github.com/mozilla/addons-server/issues/4966 -- I don't know if that fix has fully landed everywhere yet.

**Comment by ValentinaPC** [comment]
> This should have been fixed in mozilla/addons-server#4966 -- I don't know if that fix has fully landed everywhere yet.

It has landed on AMO-dev, so we are good.
Thanks for you reply!
